src/services/block.service.ts

```diff
@@ -116,6 +125,96 @@ class BlockService {
       .getRawOne();
     return Number(height);
   }
+
+  async getStatisticsBlocks(
+    offset: number,
+    limit: number,
+    orderBy: keyof BlockEntity,
+    orderDirection: 'DESC' | 'ASC',
+    period: TPeriod,
+  ): Promise<BlockEntity[]> {
+    const query: FindManyOptions<BlockEntity> = {
+      // skip: offset,
+      // take: limit,
+      order: {
+        [orderBy]: orderDirection,
+      },
+    };
+    if (period) {
+      const fromTime = getStartPoint(period) / 1000;
+      query.where = {
+        timestamp: Between(fromTime, new Date().getTime() / 1000),
+      };
+    }
+    if (offset) {
+      query.skip = offset;
+    }
+    if (limit) {
+      query.take = limit;
+    }
+    if (limit) {
+      const statsInfo = await this.getRepository().find(query);
+      return statsInfo;
+    }
+    const count = await this.getRepository().count(query);
+    const take = 500;
+    const skip = Math.round(count / take);
+    let data = [];
+    // get statistics data limit 500 for chart
+    if (count <= take || skip < 2) {
+      const statsInfo = await this.getRepository().find(query);
+      return statsInfo;
+    } else {
+      let index = 0;
+      for (let i = 0; i <= count; i += skip) {
+        const item = await this.getRepository().find({
+          take: 1,
+          skip: skip * index,
+        });
+        index += 1;
+        if (item && item.length) {
+          data = data.concat(item);
+        }
+      }
+      return data;
+    }
+  }
+  async getAverageBlockSizeStatistics(
+    period: TPeriod,
+    granularity: TGranularity,
+  ) {
+    const { groupBy, whereSqlText } = getSqlTextByPeriodGranularity(
+      period,
+      granularity,
+    );
+    const data = await this.getRepository()
+      .createQueryBuilder('block')
+      .select([])
+      .addSelect(groupBy, 'time')
+      .addSelect('AVG(size)', 'size')
+      .where(whereSqlText)
+      .groupBy(groupBy)
+      .getRawMany();
+    return data;
+  }
+
+  async getBlocksInfo(
+    sqlQuery: string,
+    period: TPeriod,
+    granularity: TGranularity,
+  ) {
+    const { groupBy, whereSqlText } = getSqlTextByPeriodGranularity(
+      period,
+      granularity,
+    );
+    return await this.getRepository()
+      .createQueryBuilder()
+      .select(groupBy, 'label')
+      .addSelect(`round(${sqlQuery}, 2)`, 'value')
```

Comment:
SQL injection vulnerability here

Reply:
This issue, It converted to func and col parameters, and I validate by yup
![image](https://user-images.githubusercontent.com/69632363/127424382-af468769-ad16-4848-98cd-4674024e20ee.png)
![image](https://user-images.githubusercontent.com/69632363/127424407-4bca9326-f809-4ab0-a844-4d9c49a14b2e.png)